src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -116,14 +116,42 @@ export abstract class BaseWindow extends Disposable implements IBaseWindow {
 	protected _lastFocusTime = Date.now(); // window is shown on creation so take current time
 	get lastFocusTime(): number { return this._lastFocusTime; }
 
+	private _maximizedWindowState: IWindowState | undefined;
+	get maximizedWindowState(): IWindowState | undefined { return this._maximizedWindowState; }
+
 	protected _win: electron.BrowserWindow | null = null;
 	get win() { return this._win; }
 	protected setWin(win: electron.BrowserWindow, options?: BrowserWindowConstructorOptions): void {
 		this._win = win;
 
 		// Window Events
-		this._register(Event.fromNodeEventEmitter(win, 'maximize')(() => this._onDidMaximize.fire()));
-		this._register(Event.fromNodeEventEmitter(win, 'unmaximize')(() => this._onDidUnmaximize.fire()));
+		this._register(Event.fromNodeEventEmitter(win, 'maximize')(() => {
```

Comment:
Why do we actually have to persist the state when the window maximizes. Can we not just compute it when `onDisplayAdded` is called in `app.ts` ad hoc? Or is that too late?

Reply:
The issue only affects maximized frameless windows and specifically when the primary display gets swapped which  happens when RDP session connects and disconnects. In all other cases the system restores the window to its rightful display without any additional work from the application.

can you expand what would computing in `OnDisplayAdded` mean ? This would imply we are computing too early when connection happens the window might not be maximized yet or too late when disconnects.

Reply:
@deepak1556 I mean, why do we need to compute and store the window state here, if we could compute it where we use it when display is added. Is that already too late?

Reply:
That would be late since we would be working with the target display rather than the source display where the window would have been maximized.

Reply:
Got it.